### PR TITLE
{tools}[GCC/8.3.0] tmux v3.1c

### PR DESCRIPTION
**Pull request by dagonzalezfo**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@dagonzalezfo Thanks a lot for your contribution!

As the failing tests indicate, there's a couple of "code style" issues that we'd like to see resolved.

To get more details, you can use `eb --check-style`, see:

```
$ eb --check-style --from-pr 12004
== temporary log file in case of crash /tmp/eb-n27qau_j/easybuild-ntsoi_7b.log

Running style check on 1 easyconfig(s)...

/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:18:8: E225 missing whitespace around operator
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:19:81: W299 trailing whitespace
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:22:13: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:22:21: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:22:24: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:22:35: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:26:4: E121 continuation line under-indented for hanging indent
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:27:17: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:27:26: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:27:29: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:27:40: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:28:19: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:28:22: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:28:33: E231 missing whitespace after ','
/tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:32:1: E124 closing bracket does not match visual indentation
[FAIL] /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb

ERROR: One or more style checks FAILED!
```

Are you OK with looking into fixing those issues?

Do let us know if something isn't clear!

**Comment by dagonzalezfo** [comment]
Hi Kenneth,

Thanks for the email.

El vie., 15 ene. 2021 19:12, Kenneth Hoste <notifications@github.com>
escribió:

> @dagonzalezfo <https://github.com/dagonzalezfo> Thanks a lot for your
> contribution!
>
> As the failing tests indicate, there's a couple of "code style" issues
> that we'd like to see resolved.
>
> To get more details, you can use eb --check-style, see:
>
> $ eb --check-style --from-pr 12004
> == temporary log file in case of crash /tmp/eb-n27qau_j/easybuild-ntsoi_7b.log
>
> Running style check on 1 easyconfig(s)...
>
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:18:8: E225 missing whitespace around operator
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:19:81: W299 trailing whitespace
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:22:13: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:22:21: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:22:24: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:22:35: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:26:4: E121 continuation line under-indented for hanging indent
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:27:17: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:27:26: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:27:29: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:27:40: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:28:19: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:28:22: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:28:33: E231 missing whitespace after ','
> /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb:32:1: E124 closing bracket does not match visual indentation
> [FAIL] /tmp/eb-n27qau_j/files_pr12004/t/tmux/tmux-3.1c-GCC-8.3.0.eb
>
> ERROR: One or more style checks FAILED
>
> !
>
> Are you OK with looking into fixing those issues?
>
Of course, thanks in advance for the information. I will do my best to
solve it,

Do let us know if something isn't clear!

**Comment by migueldiascosta** [comment]
@dagonzalezfo I think this could be moved down to `GCCcore` (see e.g. https://github.com/easybuilders/easybuild/issues/397), and both `ncurses` and `libevent` are available as easybuild dependencies, so I'd suggest something like

```diff
--- tmux-3.1c-GCC-8.3.0.eb	2021-01-26 12:35:01.988484820 +0800
+++ tmux-3.1c-GCCcore-8.3.0.eb	2021-01-26 12:43:51.979009778 +0800
@@ -11,24 +11,22 @@
 homepage = 'https://github.com/tmux/tmux'
 description = """tmux is a terminal multiplexer. It lets you switch easily between several programs in one terminal."""
 
-toolchain = {'name': 'GCC',  'version': '8.3.0'}
+toolchain = {'name': 'GCCcore',  'version': '8.3.0'}
 
 sources = ['https://github.com/tmux/tmux/releases/download/%(version)s/%(name)s-%(version)s.tar.gz']
 checksums = ['918f7220447bef33a1902d4faff05317afd9db4ae1c9971bef5c787ac6c88386']
 
 dependencies = [
-    ('Bison', '3.3.2', '', ('GCCcore', '8.3.0')),
+    ('Bison', '3.3.2'),
+    ('ncurses', '6.1'),
+    ('libevent', '2.1.11'),
 ]
 
 builddependencies = [
-    ('Autotools',  '20180311',  '',  ('GCCcore',  '8.3.0')),
-    ('pkg-config', '0.29.2', '', ('GCCcore', '8.3.0')),
-    ('M4',  '1.4.18', '', ('GCCcore', '8.3.0')),
-]
-
-osdependencies = [
-    ('ncurses-base'),
-    ('libevent-2.1-6'),
+    ('Autotools',  '20180311'),
+    ('binutils', '2.32'),
+    ('pkg-config', '0.29.2'),
+    ('M4',  '1.4.18'),
 ]
 
 sanity_check_paths = {
```

File: easybuild/easyconfigs/t/tmux/tmux-3.1c-GCCcore-8.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+name = 'tmux'
+version = '3.1c'
+
+homepage = 'https://github.com/tmux/tmux'
+description = """tmux is a terminal multiplexer. It lets you switch easily between several programs in one terminal."""
+
+toolchain = {'name': 'GCCcore',  'version': '8.3.0'}
+
+sources = ['https://github.com/tmux/tmux/releases/download/%(version)s/%(name)s-%(version)s.tar.gz']
```
Comment (Micket) [inline easybuild/easyconfigs/t/tmux/tmux-3.1c-GCCcore-8.3.0.eb:16]:
```suggestion
source_urls = ['https://github.com/tmux/tmux/releases/download/%(version)s/']
sources = [SOURCE_TAR_GZ]
```

**Review by Micket — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3519.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/efbf76780965c4627850ab180a308e86 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2665.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/e4bf8333f9cd7ba7ab06346e3ac6d039 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @dagonzalezfo!

**Comment by dagonzalezfo** [comment]
Thank all of you for your suggestions.

**Pull request merged**